### PR TITLE
Provide source artifacts distribution of PY/PC for building plugins

### PR DESCRIPTION
**Pull request by sproshev**
Update gradle-intellij-plugin to support sources for plugins for PyCharm Community and Professional Editions.
Don't hesitate to correct my code style :)

### CODE REVIEW

**Comment by zolotov** [comment]
lgtm. does it work? :)

**Comment by sproshev** [comment]
Sources are attached to IDE:


Sources are attached to plugin:

File: src/main/kotlin/org/jetbrains/intellij/dependency/PluginDependencyManager.kt [hunk 1]
```diff
@@ -106,7 +99,8 @@ open class PluginDependencyManager @Inject constructor(
     private fun registerRepositoryIfNeeded(project: Project, plugin: PluginDependency) {
         if (ivyArtifactRepository == null) {
             ivyArtifactRepository = project.repositories.ivy {
-                it.ivyPattern("$cacheDirectoryPath/[organisation]/[module]-[revision].[ext]") // ivy xml
+                val ivyFileSuffix = plugin.getFqn().substring("${plugin.id}-${plugin.version}".length)
```
Comment (zolotov) [inline src/main/kotlin/org/jetbrains/intellij/dependency/PluginDependencyManager.kt:102]: I think it won't work. for `plugin-123-2.xml` prefix would be equal to `-2.xml` and according to pattern the following file will be expected: `$cacheDirectoryPath/[organisation]/[module]-[revision]-2.xml.xml`

File: src/main/kotlin/org/jetbrains/intellij/dependency/PluginDependencyManager.kt [hunk 2]
```diff
@@ -148,7 +142,8 @@ open class PluginDependencyManager @Inject constructor(
                 addArtifact(IntellijIvyArtifact.createDirectoryDependency(it, configuration.name, baseDir, groupId))
             }
             ideaDependency?.sources?.takeIf { plugin.builtin }?.let {
-                val artifact = IntellijIvyArtifact(it, "ideaIC", "jar", "sources", "sources")
+                val name = if (isDependencyOnPyCharm(ideaDependency)) "pycharmPC" else "ideaIC"
```
Comment (zolotov) [inline src/main/kotlin/org/jetbrains/intellij/dependency/PluginDependencyManager.kt:145]: I think it's ok, but for even better experience it makes sense to download pycharm sources and attach them to Python plugin dependency even if `ideaDependency` is IntelliJ

**Review by zolotov — CHANGES_REQUESTED** [review]
please fix suffix, see IdeaDependencyManager#register

File: src/main/kotlin/org/jetbrains/intellij/dependency/PluginDependencyManager.kt (hunk 1 above)
Comment (sproshev) [reply src/main/kotlin/org/jetbrains/intellij/dependency/PluginDependencyManager.kt:102]: `plugin.getFqn()` does not have `.xml` as a suffix:
pluginFqn

**Review by zolotov — APPROVED** [review]

**Pull request merged**